### PR TITLE
add `expansion_strategy` keyword to `qml.draw_mpl`

### PR DESCRIPTION
**Pull request by albi3ro**
This PR adds the `expansion_strategy` keyword to `draw_mpl`.

Screen Shot 2022-03-03 at 10 30 09 AM

### CODE REVIEW

File: pennylane/transforms/draw.py [hunk 1]
```diff
@@ -540,7 +552,13 @@ def circuit2(x, y):
 
     @wraps(qnode)
     def wrapper(*args, **kwargs_qnode):
-        qnode.construct(args, kwargs_qnode)
+        original_expansion_strategy = getattr(qnode, "expansion_strategy", None)
+
+        try:
+            qnode.expansion_strategy = expansion_strategy or original_expansion_strategy
+            tapes = qnode.construct(args, kwargs_qnode)
```
Comment (albi3ro) [inline pennylane/transforms/draw.py:559]:
```suggestion
            qnode.construct(args, kwargs_qnode)
```

File: pennylane/transforms/draw.py [hunk 2]
```diff
@@ -540,7 +552,13 @@ def circuit2(x, y):
...
     @wraps(qnode)
     def wrapper(*args, **kwargs_qnode):
-        qnode.construct(args, kwargs_qnode)
+        original_expansion_strategy = getattr(qnode, "expansion_strategy", None)
+
+        try:
+            qnode.expansion_strategy = expansion_strategy or original_expansion_strategy
+            qnode.construct(args, kwargs_qnode)
+        finally:
+            qnode.expansion_strategy = original_expansion_strategy
```
Comment (DSGuala) [inline pennylane/transforms/draw.py:561]: I didn't know about `finally`. Cool! Is it necessary here or is this best practices?

File: tests/transforms/test_draw_mpl.py [hunk 3]
```diff
@@ -68,6 +68,21 @@ def test_standard_use():
     plt.close()
 
 
+@pytest.mark.parametrize(
+    "strategy, initial_strategy, n_patches", [("gradient", "device", 4), ("device", "gradient", 9)]
+)
+def test_expansion_strategy(strategy, initial_strategy, n_patches):
+    @qml.qnode(qml.device("default.qubit", wires=range(2)), expansion_strategy=initial_strategy)
```
Comment (DSGuala) [inline tests/transforms/test_draw_mpl.py:75]: Great! I hadn't thought about having different expansion strategies in the device and drawing.

**Review by DSGuala — APPROVED** [review]
Looks great! If I understand correctly, we're changing the qnode expansion strategy, preparing what will be drawn, then changing the qnode expansion strategy back to the original?

File: pennylane/transforms/draw.py (hunk 2 above)
Comment (albi3ro) [reply pennylane/transforms/draw.py:561]: honestly, this is just copying what the old draw method did...

**Pull request merged**